internal/component/loki/process/process_test.go

```diff
@@ -401,6 +407,8 @@ stage.static_labels {
 }
 
 func TestDeadlockWithFrequentUpdates(t *testing.T) {
```

Comment:
This test became a bit harder to follow now. Maybe we could simplify it somehow?

Reply:
No worries, I refactored it. I think it looks better now?